utils/utils.go

```diff
@@ -158,28 +163,55 @@ func GetRandomPortInRange(portRange string) uint32 {
 	}
 
 	mathrand.Seed(time.Now().UnixNano())
-	locHolder := mathrand.Intn(len(possible))
 
-	if len(possible[locHolder]) == 1 {
-		bindPort = uint32(possible[locHolder][0])
-	} else if len(possible[locHolder]) == 2 {
-		bindPort = uint32(mathrand.Intn(int(possible[locHolder][1]-possible[locHolder][0])) + int(possible[locHolder][0]))
-	}
+	if len(authPorts) > 0 {
+		bindPort = authPorts[mathrand.Intn(len(authPorts))]
```

Comment:
I don't think we want to randomly assign a requested port. The issue with that is if I permit 2 ports and pass two forwarding requests, sish could swap the tunnels that should be hosting each port.

Reply:
So you would block the process in this case?

Reply:
bump